pages/studentPortal/web/React/assignments/tailwindAssignment.tsx

```diff
@@ -0,0 +1,184 @@
+
+import { useAuth } from "../../../../../lib/authContext";
+import {
+  assignmentsSelector,
+  setUserAssignments,
+} from "../../../../../redux/assignmentsSlice";
+
+export enum Stage {
+  INCOMPLETE,
+  SUBMITTED,
+  COMPLETED,
+}
+
+export type AssignmentTemplateProps = {
+  incompleteStage: AssignmentComponentData[];
+  submittedStage: AssignmentComponentData[];
+  assignmentId: string;
+};
+
+const TailwindAssignment = ({
+  incompleteStage,
+  submittedStage,
+  assignmentId,
+}: AssignmentTemplateProps) => {
+  const [stage, setStage] = useState(0);
+  const { user } = useAuth();
+
+  const dispatch = useDispatch();
+  const { userAssignments } = useSelector(assignmentsSelector);
+
+  const deployCurrentStage = (assignment: UserAssignmentSubmissionsData) => {
+    if (assignment.review_link != null) {
+      setStage(Stage.COMPLETED);
+    } else if (assignment.submission_link) {
+      setStage(Stage.SUBMITTED);
+    } else {
+      setStage(Stage.INCOMPLETE);
+    }
+  };
+
+  const { loading: userAssignmentsLoading } =
+    useQuery<FetchUserAssignmentSubmissionsDataResponse>(
+      FETCH_USER_ASSIGNMENT_SUBMISSIONS,
+      {
+        variables: {
+          user_id: user.uid,
+          assignmentId: assignmentId,
+        },
+
+        onCompleted: (data: FetchUserAssignmentSubmissionsDataResponse) => {
+          if (data.user_assignment_submissions.length > 0) {
+            dispatch(setUserAssignments(data.user_assignment_submissions));
+            deployCurrentStage(data.user_assignment_submissions[0]);
+          }
+        },
+      }
+    );
+
+  const completedStage: AssignmentComponentData[] = [
+    {
+      component: "loom-video",
+      text: "Your assignment has been reviewed! Watch the video below for feedback:",
+      videoId: userAssignments.length > 0 ? userAssignments[0].review_link : "",
+    },
+  ];
+
+  const handleContinue = () => {
+    router.push("/studentPortal/web/React/components");
+  };
+  return (
+    <>
+      <div className="flex flex-col m-8 space-y-4 sm:px-12">
+        {userAssignmentsLoading ? (
+          <div>Loading...</div>
+        ) : stage === Stage.INCOMPLETE ? (
+          incompleteStage.map((it: AssignmentComponentData, index) => (
+            <AssignmentComponent key={index} data={it} />
+          ))
+        ) : stage === Stage.SUBMITTED ? (
+          submittedStage.map((it: AssignmentComponentData, index) => (
+            <AssignmentComponent key={index} data={it} />
+          ))
+        ) : stage === Stage.COMPLETED ? (
+          completedStage.map((it: AssignmentComponentData, index) => (
+            <AssignmentComponent key={index} data={it} />
+          ))
+        ) : null}
+        <div className="flex my-8 mr-8 sm:justify-end">
+          <Button onClick={handleContinue} label="Continue" />
+        </div>
+      </div>
+    </>
+  );
+};
+export async function getServerSideProps() {
+  const assignmentId = "2cf9156a-4f6f-452d-b09a-2c54f19a7b40";
```

Comment:
this assignmentId is the same as the other assignment

create a new assignment in the coding_assignments table and name it Tailwind Assignment. You can then paste the ID that hasura generates over here